### PR TITLE
Prevent extra volume mount/umount on btrfs

### LINKED ISSUE #2529: kiwi 10 fails to umount some subvolumes
## Problem description

https://build.opensuse.org/package/live_build_log/openSUSE:Factory/openSUSE-MicroOS:kvm-and-xen-sdboot/images/x86_64

Has errors like this:

```
[  104s] [ DEBUG   ]: 16:21:35 | EXEC: [umount /var/tmp/kiwi_volumes.smcpb8p1]
[  104s] [ DEBUG   ]: 16:21:35 | EXEC: Failed with stderr: umount: /var/tmp/kiwi_volumes.smcpb8p1: target is busy.
[  104s] , stdout: (no output on stdout)
[  104s] [ WARNING ]: 16:21:35 | 0 umount of /var/tmp/kiwi_volumes.smcpb8p1 failed with: umount: stderr: umount: /var/tmp/kiwi_volumes.smcpb8p1: target is busy.
[  104s] , stdout: (no output on stdout)
```

That does not happen with python3-kiwi-9.25.22-2.2

Comparing the logs one can see that the new version does not umount certain subvolume underneath `/@/.snapshots/1/snapshot`, eg there's a a line

```
[   74s] [ DEBUG   ]: 12:11:46 | EXEC: [mount -o subvol=@/home /dev/loop0p3 /var/tmp/kiwi_volumes.tis_hfxv/@/.snapshots/1/snapshot/home]
```
but no matching umount. In kiwi9 it did umount.

## Steps to reproduce the behaviour

Build https://build.opensuse.org/package/show/openSUSE:Factory/openSUSE-MicroOS using profile kvm-and-xen-sdboot
Reproducible locally.

## OS and Software information

* KIWI version: 10.0.10-2.1
* Operating system host version: Tumbleweed
* Operating system target version: Tumbleweed

### PR DESCRIPTION
**Pull request by schaefi**
For setting up the read-only property an extra mount of the btrfs sub-volumes was issued. However, all volumes are mounted at that time. Thus it's not required to mount them again, resulting in a busy state because of the auto-snapshot mounts which does not get umounted and keeps a busy state until the lazy umount kicks in. This Fixes #2529

### CODE REVIEW

**Comment by lnussel** [comment]
It fixes the busy issue but the real problem is still lurking. `sync_target` is actually the root subvolume rather than snapshot 1. therefore the property ends up set on the wrong subvolume

**Comment by lnussel** [comment]
I think it needs `sync_target = self.get_mountpoint()` instead of `sync_target = self.mountpoint`

**Comment by schaefi** [comment]
yes it this makes sense. I'll give it a test

**Comment by lnussel** [comment]
yup, with self.get_mountpoint() it works

**Comment by schaefi** [comment]
@lnussel also tested this successfully if the bootloader is not custom. To make sure there are no regressions by this change I will create a Staging build after this one has been merged.

Thanks

**Comment by schaefi** [comment]
waiting for staging builds to complete

**Comment by schaefi** [comment]
integration tests looks good

**Comment by lnussel** [comment]
any ETA? currently systemd-boot based images in Factory fail

**Comment by schaefi** [comment]
> any ETA? currently systemd-boot based images in Factory fail

I was waiting for an approval from at least one involved person. I think we are good and I have a version in Staging and all integration test builds looks good. If you or Neal agrees I'll continue and make a submission today.

**Comment by Conan-Kudo** [comment]
Won't this mess up non-Btrfs stuff (e.g. LVM disks)? I vaguely recall that we have this specifically to ensure that everything cleans up properly...

**Comment by schaefi** [comment]
> Won't this mess up non-Btrfs stuff (e.g. LVM disks)? I vaguely recall that we have this specifically to ensure that everything cleans up properly...

I was also concerned about the other volume manager but due to the refactoring into the context managers we have proper cleanup which allowed to cleanup some places of explicit mount/umount calls. There is an integration test for testing LVM and I did a closer look to that build to check if there are any regressions. I could not find one.

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**
